include/up-cpp/uri/builder/BuildUAuthority.h

```diff
@@ -95,54 +96,53 @@ namespace uprotocol::uri {
         }
-    
+
         /**
          * set authority IP address. if address is empty or blank, an error will be logged and the address will not be set
          * wrong IP address format will be ignored
          * @param address std::string
-         * @return 
+         * @return
          */
         auto setIp(const std::string &address) -> BuildUAuthority & {
+            return setIp(IpAddress(address));
+        }
+
+        /**
+         * set authority IP address. if address is empty or blank, an error will be logged and the address will not be set
+         * wrong IP address format will be ignored
+         * @param address IpAddress
+         * @return
+         */
+        auto setIp(const IpAddress &address) -> BuildUAuthority & {
             if (authority_.has_ip() && !authority_.ip().empty()) {
-                spdlog::error("UAutority already has ip set {}. Ignoring setIp()", authority_.ip());
+                spdlog::error("UAuthority already has ip set {}. Ignoring setIp()", authority_.ip());
                 return *this;
             }
-            std::string m_address = isBlank(address) ? "" : address;
-            
-            char ip_char[INET6_ADDRSTRLEN + 1];
-            memset(ip_char, 0, INET6_ADDRSTRLEN + 1);
-            if (in6_addr ipv6{}; inet_pton(AF_INET6, m_address.c_str(), &ipv6) == 1) {
-                if (inet_ntop(AF_INET6,
-                              &ipv6,
-                              ip_char,
-                              INET6_ADDRSTRLEN) != nullptr) {
-                    authority_.set_ip(ip_char);
-                }
-            } else if (struct in_addr ipv4{}; inet_pton(AF_INET, m_address.c_str(), &ipv4) == 1) {
-                if (inet_ntop(AF_INET,
-                              &ipv4, 
-                              ip_char, 
-                              INET_ADDRSTRLEN) != nullptr) {
-                    authority_.set_ip(ip_char);
-                } else {
-                    spdlog::error<std::string_view>("UAutority address is not a valid IPV4 address. Ignoring setIp()");
-                    return *this;
-                }
-            } else {
-                spdlog::error<std::string_view>("UAutority address is not a valid IP address. Ignoring setIp()");
+
+            if (address.getType() == IpAddress::Type::Invalid) {
+                spdlog::error<std::string_view>("UAuthority address is not a valid IP address. Ignoring setIp()");
+                // Note: setting an empty string here will allow the micro
+                //       serializer to detect that something was wrong instead
+                //       of thinking it has been asked to serialize a Local
+                //       authority.
+                authority_.set_ip("");
+                return *this;
             }
+
+            authority_.set_ip(address.getBytesString());
+
             return *this;
         }
-        
+
         auto setId(const std::string &id) -> BuildUAuthority & {
```

Comment:
consider adding comments to explain the purpose of this method and any constraints related to the ID

Reply:
#85 filed to track this request.